pulp_docker/app/tasks/sync_stages.py

```diff
@@ -43,25 +34,130 @@ def __init__(self, remote):
 
     async def run(self):
         """
-        Build and emit `DeclarativeContent` for each Tag.
+        DockerFirstStage.
+        """
+        future_manifests = []
+        tag_list = []
+        to_download = []
+        man_dcs = {}
+        total_blobs = []
+
+        with ProgressBar(message='Downloading tag list for the repo', total=1) as pb:
+            relative_url = '/v2/{name}/tags/list'.format(name=self.remote.namespaced_upstream_name)
+            tag_list_url = urljoin(self.remote.url, relative_url)
+            list_downloader = self.remote.get_downloader(tag_list_url)
+            await list_downloader.run()
+
+            with open(list_downloader.path) as tags_raw:
+                tags_dict = json.loads(tags_raw.read())
+                tag_list = tags_dict['tags']
+
+            pb.increment()
+
+        with ProgressBar(message='Creating Download requests for Tags', total=len(tag_list)) as pb:
+            for tag_name in tag_list:
+                relative_url = '/v2/{name}/manifests/{tag}'.format(
+                    name=self.remote.namespaced_upstream_name,
+                    tag=tag_name,
+                )
+                url = urljoin(self.remote.url, relative_url)
+                downloader = self.remote.get_downloader(url=url)
+                to_download.append(downloader.run(extra_data={'headers': V2_ACCEPT_HEADERS}))
+                pb.increment()
+
+        pb1 = ProgressBar(message='Parsed SchemaV2 Tags', state='running')
+        pb2 = ProgressBar(message='Parsing Manifest List Tags', state='running')
+        pb3 = ProgressBar(message='Parsing Manifests Tags', state='running')
+        global pb4
+        pb4 = ProgressBar(message='Parsing Blobs', state='running')
+        pb5 = ProgressBar(message='Parsing Manifests', state='running')
```

Comment:
I like how you did these progress bars. It nicely separates the workflow we need to follow from the workflow the user would want to see.